cloudant.js

```diff
@@ -29,67 +29,35 @@ var reconfigure = require('./lib/reconfigure.js');
 function Cloudant(options, callback) {
   debug('Initialize', options);
 
-  // Save the username and password for potential conversion to cookie auth.
-  var login = reconfigure.getOptions(options);
-
-  // Convert the credentials into a URL that will work for cloudant. The
-  // credentials object will become squashed into a string, which is fine
-  // except for the .cookie option.
-  var cookie = options.cookie;
-
-  var pkg = require('./package.json');
-  var requestDefaults = {
-    gzip: true,
-    headers: {
-      // set library UA header
-      'User-Agent': `nodejs-cloudant/${pkg.version} (Node.js ${process.version})`
-    },
-    jar: false
-  };
-  var theurl = null;
-  if (typeof options === 'object') {
-    if (options.requestDefaults) {
-      requestDefaults = options.requestDefaults;
-    }
-    theurl = reconfigure(options);
-  } else {
-    theurl = reconfigure({ url: options });
+  if (typeof options !== 'object') {
+    options = { url: options };
   }
+
+  var theurl = reconfigure(options);
   if (theurl === null) {
+    var err = new Error('Invalid URL');
     if (callback) {
-      return callback('invalid url', null); // eslint-disable-line standard/no-callback-literal
+      return callback(err);
     } else {
-      throw (new Error('invalid url'));
+      throw (err);
     }
   }
 
-  // keep connections alive by default
-  if (requestDefaults && !requestDefaults.agent) {
-    var protocol = (theurl.match(/^https/)) ? require('https') : require('http');
-    var agent = new protocol.Agent({
-      keepAlive: true,
-      keepAliveMsecs: 30000,
-      maxSockets: 6
-    });
-    requestDefaults.agent = agent;
+  if (theurl.match(/^http:/)) {
+    options.https = false;
   }
 
-  // plugin a request library
-  var plugin = null;
-  if (options.plugin) {
-    options.requestDefaults = requestDefaults;
-    if (typeof options.plugin === 'string') {
-      var plugintype = options.plugin || 'default';
-      debug('Using the "' + plugintype + '" plugin');
-      plugin = require('./plugins/' + plugintype)(options);
-    } else if (typeof options.plugin === 'function') {
-      debug('Using a custom plugin');
-      plugin = options.plugin;
-    }
-  }
+  // build request client
+  var cloudantClient = new Client(options);
+  var cloudantRequest = function(req, callback) {
+    return cloudantClient.request(req, callback);
+  };
 
-  debug('Create underlying Nano instance, options=%j requestDefaults=%j', options, requestDefaults);
```

Comment:
We should keep this debug message, if it still makes sense

Reply:
Yup, will add.